drivers/dma/dma_stm32.c

```diff
@@ -232,13 +224,8 @@ static int dma_stm32_get_periph_increment(enum dma_addr_adj increment,
 	return 0;
 }
 
-#ifdef CONFIG_DMAMUX_STM32
-int dma_stm32_configure(struct device *dev, uint32_t id,
-			       struct dma_config *config)
-#else
-static int dma_stm32_configure(struct device *dev, uint32_t id,
-			       struct dma_config *config)
-#endif /* CONFIG_DMAMUX_STM32 */
+DMA_STM32_EXPORT_DMAMUX int dma_stm32_configure(struct device *dev, uint32_t id,
```

Comment:
I think there should be a more generic symbol here, like `DMA_STM32_EXPORT`. When I saw this symbol first time, I thought it's only used for the ones with DMAMUX features. What do you think?

Reply:
Well the name was based on the original `#ifdef CONFIG_DMAMUX_STM32` . But now I even wonder if it even makes sense to add all that code just to add/remove `static` to the function. I don't think it would be a problem to just always leave away the `static`?

Reply:
At least for me it's a coding habit that trying to add `static` to all the functions, making symbols less exposed in the kernel. Yes I also think this is unnecessary since we've already named the function with module name as prefix. While this is a coding rule as I remember in the Linux kernel. Zephyr is Linux's little brother, and it also inherits many Linux's coding style.

Reply:
Yeah static makes sense, no question about it, but these functions are kind of a public API so I think we can remove the DMASTM32_EXPORT_DMAMUX and just always "export" them, makes the code clearer I think.

Reply:
For the moment I just removed the `static` , but I think if we refactor the dmamux driver we probably can use the dma driver struct to get the functions. But I think that should be done in a different PR.

Reply:
@cybertale Can we put this on "resolved" too ? Maybe relook at it when refactoring the dmamux driver ?

Reply:
Of course this way it will work, but I don't know if this is acceptable. This idea of remove some existed static... I think we need some upstream maintainer's opinion like @erwango 's about this. 
This is not a big problem, should be resolved soon. Sorry I don't have much free time in work days, havn't look all your commits, but I 'll try these two days. Changes I've seen so far so good!

Reply:
I have no problem putting the defines back in, maybe with a different name like maybe DMA_STM32_EXPORT_API  ?

Reply:
@cybertale is this an acceptable solution? I think the `DMA_STM32_EXPORT_API` name should be clear enough for everybody and the code is more readable than with the whole ``#ifdef #else #endif``

Reply:
@cybertale and @erwango is the use of `DMA_STM32_EXPORT_API` OK, or should that be done in some other way?

Reply:
Yes, this is acceptable for me.